onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -72,6 +73,7 @@ std::vector<std::unique_ptr<RewriteRule>> GenerateRewriteRules(
       rules.push_back(std::make_unique<CastElimination>());
       rules.push_back(std::make_unique<NoopElimination>());
       rules.push_back(std::make_unique<DivMulFusion>());
+      rules.push_back(std::make_unique<GraphOutputDeduplication>());
       rules.push_back(std::make_unique<FuseReluClip>());
```

Comment:
I really don't think ORT should be changing the output names at model load time. Whilst in your specific use case you might expect that, I don't know that you can say that for users in general.

What's the reason this can't be handled at export time?

Reply:
agree. we don't want to modify output name on user model. It's better to be handled in exporter.